lib/main/google/olc/olc.c

```diff
@@ -104,10 +101,10 @@ int OLC_Encode(const OLC_LatLon* location, size_t length, char* code,
 
   // Multiply values by their precision and convert to positive without any
   // floating point operations.
-  long long int lat_val = kLatMaxDegrees * 2.5e7;
-  long long int lng_val = kLonMaxDegrees * 8.192e6;
-  lat_val += latitude * 2.5e7;
-  lng_val += longitude * 8.192e6;
+  long long int lat_val = kLatMaxDegrees * (double)2.5e7;
```

Comment:
`2.5e7` IS double literal. Explicit cast is ugly. And this is IMO not a case of double promotion  kLatMaxDegrees  is already double, so it is not promoted

Reply:
```./lib/main/google/olc/olc.c: In function 'OLC_Encode':
./lib/main/google/olc/olc.c:104:42: error: implicit conversion from 'float' to 'double' to match other operand of binary expression [-Werror=double-promotion]
  104 |   long long int lat_val = kLatMaxDegrees * 2.5e7;```